src/diebold_mariano.jl

```diff
@@ -0,0 +1,77 @@
+# diebold_mariano.jl
+# Diebold-Mariano
+#
+# Copyright (C) 2020   Guilherme Bodin
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+export DieboldMarianoTest
+
+struct DieboldMarianoTest <: TTest
+    n::Integer     # number of observations
+    df::Integer    # degrees of freedom
+    t::Real        # test statistic
+end
+
+"""
+    DieboldMarianoTest(x::AbstractVector{<:Real}, y::AbstractVector{<:Real})
+
+Perform the modified Diebold-Mariano test proposed by Harvey, Leybourne and Newbold of the null 
+hypothesis that the two methods have the same forecast accuracy. `g` is the loss function described
```

Comment:
`g` is mentioned here but isn't included in the signature above. Also it would be more informative IMO to call it `loss`, since it's specifying a loss function.